### PR TITLE
Add Roslyn analyzer to check results are used

### LINKED ISSUE #2442: Make ORM use more foolproof
At the moment it's easy to drop an updated entity and fail to use it in subsequent operations. This is because we have immutable record objects.

I suggest we update methods of the form `T UpdateEntity(T entity, …)` to take a `ref` parameter instead `UpdateEntity(ref T entity, …)`, so that the caller cannot fail to use the updated entity.

This will also make explicit operations which update the entity state.

### PR DESCRIPTION
**Pull request by Porges**
As seen in #2441, it is easy to drop return values of updated entities accidentally.

This PR adds a Roslyn Analyzer which will detect when return values are unused. To explicitly ignore a value you can drop it with `_ = …;`

Closes #2442.

### CODE REVIEW

File: src/ApiService/ApiService/Functions/TimerRepro.cs [hunk 1]
```diff
@@ -14,20 +14,25 @@ public class TimerRepro {
...
+                _ = expiredVmIds.Add(repro.VmId);
+                // ignoring result: value not used later
+                _ = await _onefuzzContext.ReproOperations.Stopping(repro);
+            }
         }
 
-        var expiredVmIds = expired.Select(repro => repro?.VmId);
-
         await foreach (var repro in _onefuzzContext.ReproOperations.SearchStates(VmStateHelper.NeedsWork)) {
-            if (await expiredVmIds.ContainsAsync(repro.VmId)) {
```
Comment (Porges) [inline src/ApiService/ApiService/Functions/TimerRepro.cs:26]: Also fixed this: this would perform the query again each time and be very slow.

**Review by stishkin — APPROVED** [review]

File: src/ApiService/ApiService/Functions/AgentEvents.cs [hunk 2]
```diff
@@ -155,7 +155,8 @@ public class AgentEvents {
             // if tasks are running on the node when it reports as Done
             // those are stopped early
             await _context.NodeOperations.MarkTasksStoppedEarly(node, error);
-            await _context.NodeOperations.ToReimage(node, done: true);
+            // result ignored: not used after this point
+            _ = await _context.NodeOperations.ToReimage(node, done: true);
```
Comment (chkeita) [inline src/ApiService/ApiService/Functions/AgentEvents.cs:159]: I propose we encapsulate this pattern in an extension method called ignore 
soemthing like 
``` C#
static class TaskExtensions
{
	public static async Task Ignore<T>(this Task<T> task) {
		_ = await task;
	}
}
```

and this call can become. The method name shows the intent so no need to add the comment 
``` suggestion
            await _context.NodeOperations.ToReimage(node, done: true).Ignore();
```

**Review by chkeita — APPROVED** [review]

File: src/ApiService/ApiService/Functions/AgentEvents.cs (hunk 2 above)
Comment (Porges) [reply src/ApiService/ApiService/Functions/AgentEvents.cs:159]: Good idea. This will make it easier to find and fix things in future as well.

Comment (Porges) [reply src/ApiService/ApiService/Functions/AgentEvents.cs:163]: At the moment I'm only using this helper in cases where we need to address/fix this in future.

**Pull request merged**